server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -0,0 +1,251 @@
+
+    public static final Setting<TimeValue> INTERVAL_SETTING = Setting.timeSetting(
+        "health.reporting.local.monitor.interval",
+        TimeValue.timeValueSeconds(30),
+        TimeValue.timeValueSeconds(10),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    private final ClusterService clusterService;
+    private final ThreadPool threadPool;
+    private final DiskCheck diskCheck;
+
+    private volatile TimeValue monitorInterval;
+    private volatile Scheduler.ScheduledCancellable scheduled;
+    private volatile IndividualNodeHealth lastReportedHealth = null;
+    private volatile boolean enabled;
+    private volatile boolean healthMetadataInitialized;
+
+    public LocalHealthMonitor(Settings settings, ClusterService clusterService, NodeService nodeService, ThreadPool threadPool) {
+        this.threadPool = threadPool;
+        this.monitorInterval = INTERVAL_SETTING.get(settings);
+        this.enabled = HealthNodeTaskExecutor.ENABLED_SETTING.get(settings);
+        this.clusterService = clusterService;
+        this.diskCheck = new DiskCheck(nodeService);
+        clusterService.addListener(this);
+        ClusterSettings clusterSettings = clusterService.getClusterSettings();
+        clusterSettings.addSettingsUpdateConsumer(INTERVAL_SETTING, this::setMonitorInterval);
+        clusterSettings.addSettingsUpdateConsumer(HealthNodeTaskExecutor.ENABLED_SETTING, this::setEnabled);
+    }
+
+    void setMonitorInterval(TimeValue monitorInterval) {
+        this.monitorInterval = monitorInterval;
+        if (scheduled != null && scheduled.cancel()) {
+            scheduleNextRunIfEnabled(new TimeValue(1));
+        }
+    }
+
+    void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        if (scheduled != null) {
+            scheduled.cancel();
+            if (enabled) {
+                scheduleNextRunIfEnabled(new TimeValue(1));
+            }
+        }
+    }
+
+    private void scheduleNextRunIfEnabled(TimeValue time) {
+        if (enabled && threadPool.scheduler().isShutdown() == false) {
+            scheduled = threadPool.schedule(this::monitorHealth, time, ThreadPool.Names.MANAGEMENT);
+        }
+    }
+
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        // Wait until every node in the cluster is upgraded to 8.4.0 or later
+        if (event.state().nodesIfRecovered().getMinNodeVersion().onOrAfter(Version.V_8_4_0)) {
+            // Wait until the health metadata is available in the cluster state
+            if (healthMetadataInitialized == false) {
+                healthMetadataInitialized = HealthMetadata.getFromClusterState(event.state()) != null;
+                if (healthMetadataInitialized) {
+                    scheduleNextRunIfEnabled(TimeValue.timeValueMillis(1));
+                }
+            } else if (newHealthNodeSelected(event)) {
+                sendHealth(lastReportedHealth);
+            }
```

Comment:
should this be in a separate PR that handles "sending of health data" ?

Reply:
I can put it in the next PR that I am working on, thinking about it kind of helped me set up the code but I can remove this and just log a line upon a change of the node's health. 

I will remove it.